src/api/handlers/channel.ts

```diff
@@ -301,6 +303,40 @@ export async function createInvite(
   return RequestManager.post(endpoints.CHANNEL_INVITES(channelID), options);
 }
 
+/** Returns an invite for the given code. */
+export function getInvite(inviteCode: string) {
+  return RequestManager.get(endpoints.INVITE(inviteCode)) as Promise<
+    InvitePayload
+  >;
+}
+
+/** Deletes an invite for the given code. Requires `MANAGE_CHANNELS` or `MANAGE_GUILD` permission */
+export async function deleteInvite(
+  channelID: string,
+  inviteCode: string,
+) {
+  const hasPerm = await botHasChannelPermissions(channelID, [
+    "MANAGE_CHANNELS",
+  ]);
+
+  if (!hasPerm) {
+    const channel = await cacheHandlers.get("channels", channelID);
+    if (!channel) throw new Error(Errors.CHANNEL_NOT_FOUND);
```

Comment:
I don't think this is needed.

Reply:
It is, else how to check bots permission?

Reply:
You are already using the `botHasChannelPermission()` function above, which retrieves the channel from the cache, you are doing that again...

Reply:
well yes the error throwing is not required

Reply:
The entire 'retrieval from cache' is not required.

Reply:
it is, how to get the guildID which is required for perm checking.